src/murmur/Meta.cpp

```diff
@@ -452,6 +457,37 @@ void MetaParams::read(QString fname) {
 		}
 	}
 
+#if defined(USE_QSSLDIFFIEHELLMANPARAMETERS)
+	if (! qsSSLDHParams.isEmpty()) {
+		QFile pem(qsSSLDHParams);
+		if (pem.open(QIODevice::ReadOnly)) {
+			dhparams = pem.readAll();
+			pem.close();
+		} else {
+			qCritical("Failed to read %s", qPrintable(qsSSLDHParams));
+		}
+	}
+
+	if (! dhparams.isEmpty()) {
+		QSslDiffieHellmanParameters qdhp = QSslDiffieHellmanParameters(dhparams);
+		if (qdhp.isValid()) {
+			qbaDHParams = dhparams;
+		} else {
+			QSet<QSslDiffieHellmanParameters::Error> errors = qdhp.errors();
```

Comment:
Same as above. Also it might be worth it to extract this functionality into a function.